### ISSUE TITLE
ProductFacade::edit() does not always export data to elasticsearch

### ISSUE THREAD

**Issue by grossmannmartin** [shopsys/shopsys#2088]
### What is happening
`ProductFacade::edit()` calls `productExportScheduler` to schedule product for immediate export to elastic 
```
$this->productExportScheduler->scheduleRowIdForImmediateExport($productToExport->getId());
```

this method store IDs meant to be exported only in memory and are properly exported after firing the response event (`kernel.response`). 
See `ProductExportSubscriber` class.

This causes troubles in non-request-response scenarios (for example in cron jobs) because those IDs are never exported, nor marked to export in the database.

It's unintuitive and developers can easily forget to call exports in their own code causing out of sync data on the frontend.

### Expected result
The ideal solution for me is to not be forced to think about such details, so the behavior should be the same whenever I edit the product from the administration or in some import crons, commands, etc.

For example, async queue could help here a lot.

**Comment by TomasLudvik** [comment]
I was just wandering, what if we started using both
https://symfony.com/doc/current/reference/events.html#kernel-terminate and
https://symfony.com/doc/current/components/console/events.html#console-events-terminate
instead of kernel.response?

**Comment by grossmannmartin** [comment]
resolved by #2917
